homeassistant/components/config/config_entries.py

```diff
@@ -483,13 +484,17 @@ async def async_matching_config_entries(
         elif not isinstance(integration_or_exc, IntegrationNotFound):
             raise integration_or_exc
 
+    # Filter out entries that don't match the type filter
+    # when only helpers are requested, also filter out entries
+    # from unknown integrations. This prevent them from showing
+    # up in the helpers UI.
     entries = [
         entry
         for entry in entries
-        if (type_filter != "helper" and entry.domain not in integrations)
+        if (type_filter != ["helper"] and entry.domain not in integrations)
```

Comment:
I've added a comment about why this is/was happening with the helper filter for now. This is not stuff that should happen like this here. Instead, we should add a parameter to this call that allows for returning this if wanted (and not magically by category).